In [153]:
import os
import pandas as pd
import statsmodels.formula.api as sm

if not os.getcwd().endswith('assignment2'):
    os.chdir(os.path.join('Unit2', 'assignment2'))
    
elantra_sales = pd.read_csv('elantra.csv')


### Problem 1
Split the data set into training (before 2012, inclusive) and testing (after 2012) sets.
How many observations are in the training set?

The split:

In [154]:
elantra_train = elantra_sales[elantra_sales['Year'] <= 2012]
elantra_test = elantra_sales[elantra_sales['Year'] > 2012]

The size of the training data:

In [155]:
len(elantra_train.index)

36

### problem 2.1
Build a linear regression model to predict monthly Elantra sales using Unemployment, CPI_all, CPI_energy and Queries as the independent variables. 
What is the model R-squared?


In [156]:
# let's have a look at the variables
elantra_sales.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 50 entries, 0 to 49
Data columns (total 7 columns):
Month           50 non-null int64
Year            50 non-null int64
ElantraSales    50 non-null int64
Unemployment    50 non-null float64
Queries         50 non-null int64
CPI_energy      50 non-null float64
CPI_all         50 non-null float64
dtypes: float64(3), int64(4)
memory usage: 2.8 KB


In [157]:
elantra_sales.head(n=4)

<div>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Month</th>
      <th>Year</th>
      <th>ElantraSales</th>
      <th>Unemployment</th>
      <th>Queries</th>
      <th>CPI_energy</th>
      <th>CPI_all</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>1</td>
      <td>2010</td>
      <td>7690</td>
      <td>9.7</td>
      <td>153</td>
      <td>213.377</td>
      <td>217.466</td>
    </tr>
    <tr>
      <th>1</th>
      <td>1</td>
      <td>2011</td>
      <td>9659</td>
      <td>9.1</td>
      <td>259</td>
      <td>229.353</td>
      <td>221.082</td>
    </tr>
    <tr>
      <th>2</th>
      <td>1</td>
      <td>2012</td>
      <td>10900</td>
      <td>8.2</td>
      <td>354</td>
      <td>244.178</td>
      <td>227.666</td>
    </tr>
    <tr>
      <th>3</th>
      <td>1</td>
      <td>2013</td>
      <td>12174</td>
      <td>7.9</td>
      <td>230</td>
      <td>242.560</td>
      <td>231.321</td>
    </tr>
  </tbody>
</table>
</div>

In [158]:
model1 = sm.ols(formula="ElantraSales ~ Unemployment + "
                        "CPI_all + CPI_energy + Queries", 
                data=elantra_train).fit()
model1.rsquared

0.42815676686082871

### problem 2.2
How many variables are significant (use p value above 0.1)

In [159]:
result.summary()

<table class="simpletable">
<caption>OLS Regression Results</caption>
<tr>
  <th>Dep. Variable:</th>      <td>ElantraSales</td>   <th>  R-squared:         </th> <td>   0.428</td>
</tr>
<tr>
  <th>Model:</th>                   <td>OLS</td>       <th>  Adj. R-squared:    </th> <td>   0.354</td>
</tr>
<tr>
  <th>Method:</th>             <td>Least Squares</td>  <th>  F-statistic:       </th> <td>   5.803</td>
</tr>
<tr>
  <th>Date:</th>             <td>Tue, 21 Mar 2017</td> <th>  Prob (F-statistic):</th>  <td>0.00132</td>
</tr>
<tr>
  <th>Time:</th>                 <td>08:37:54</td>     <th>  Log-Likelihood:    </th> <td> -339.99</td>
</tr>
<tr>
  <th>No. Observations:</th>      <td>    36</td>      <th>  AIC:               </th> <td>   690.0</td>
</tr>
<tr>
  <th>Df Residuals:</th>          <td>    31</td>      <th>  BIC:               </th> <td>   697.9</td>
</tr>
<tr>
  <th>Df Model:</th>              <td>     4</td>      <th>                     </th>     <td> </td>   
</tr>
<tr>
  <th>Covariance Type:</th>      <td>nonrobust</td>    <th>                     </th>     <td> </td>   
</tr>
</table>
<table class="simpletable">
<tr>
        <td></td>          <th>coef</th>     <th>std err</th>      <th>t</th>      <th>P>|t|</th>  <th>[0.025</th>    <th>0.975]</th>  
</tr>
<tr>
  <th>Intercept</th>    <td> 9.539e+04</td> <td> 1.71e+05</td> <td>    0.559</td> <td> 0.580</td> <td>-2.53e+05</td> <td> 4.43e+05</td>
</tr>
<tr>
  <th>Unemployment</th> <td>-3179.8996</td> <td> 3610.262</td> <td>   -0.881</td> <td> 0.385</td> <td>-1.05e+04</td> <td> 4183.279</td>
</tr>
<tr>
  <th>CPI_all</th>      <td> -297.6456</td> <td>  704.837</td> <td>   -0.422</td> <td> 0.676</td> <td>-1735.169</td> <td> 1139.878</td>
</tr>
<tr>
  <th>CPI_energy</th>   <td>   38.5060</td> <td>  109.601</td> <td>    0.351</td> <td> 0.728</td> <td> -185.027</td> <td>  262.039</td>
</tr>
<tr>
  <th>Queries</th>      <td>   19.0297</td> <td>   11.259</td> <td>    1.690</td> <td> 0.101</td> <td>   -3.933</td> <td>   41.992</td>
</tr>
</table>
<table class="simpletable">
<tr>
  <th>Omnibus:</th>       <td> 1.210</td> <th>  Durbin-Watson:     </th> <td>   1.190</td>
</tr>
<tr>
  <th>Prob(Omnibus):</th> <td> 0.546</td> <th>  Jarque-Bera (JB):  </th> <td>   0.947</td>
</tr>
<tr>
  <th>Skew:</th>          <td> 0.390</td> <th>  Prob(JB):          </th> <td>   0.623</td>
</tr>
<tr>
  <th>Kurtosis:</th>      <td> 2.845</td> <th>  Cond. No.          </th> <td>1.32e+05</td>
</tr>
</table>

No variable has a significant p-value (<0.1)

### Problem 2.3
What is the coefficient of the Unemployment variable?

In [160]:
model1.params

Intercept       95385.363597
Unemployment    -3179.899573
CPI_all          -297.645626
CPI_energy         38.506037
Queries            19.029683
dtype: float64

### Problem 3.1
Incorporate data from months varible. What is the new R-Squared?

In [161]:
model2 = sm.ols(formula="ElantraSales ~ Unemployment + CPI_all +"
                        " CPI_energy + Queries + Month", 
                data=elantra_train).fit()
model2.summary()

<table class="simpletable">
<caption>OLS Regression Results</caption>
<tr>
  <th>Dep. Variable:</th>      <td>ElantraSales</td>   <th>  R-squared:         </th> <td>   0.434</td>
</tr>
<tr>
  <th>Model:</th>                   <td>OLS</td>       <th>  Adj. R-squared:    </th> <td>   0.340</td>
</tr>
<tr>
  <th>Method:</th>             <td>Least Squares</td>  <th>  F-statistic:       </th> <td>   4.609</td>
</tr>
<tr>
  <th>Date:</th>             <td>Tue, 21 Mar 2017</td> <th>  Prob (F-statistic):</th>  <td>0.00308</td>
</tr>
<tr>
  <th>Time:</th>                 <td>08:37:55</td>     <th>  Log-Likelihood:    </th> <td> -339.79</td>
</tr>
<tr>
  <th>No. Observations:</th>      <td>    36</td>      <th>  AIC:               </th> <td>   691.6</td>
</tr>
<tr>
  <th>Df Residuals:</th>          <td>    30</td>      <th>  BIC:               </th> <td>   701.1</td>
</tr>
<tr>
  <th>Df Model:</th>              <td>     5</td>      <th>                     </th>     <td> </td>   
</tr>
<tr>
  <th>Covariance Type:</th>      <td>nonrobust</td>    <th>                     </th>     <td> </td>   
</tr>
</table>
<table class="simpletable">
<tr>
        <td></td>          <th>coef</th>     <th>std err</th>      <th>t</th>      <th>P>|t|</th>  <th>[0.025</th>    <th>0.975]</th>  
</tr>
<tr>
  <th>Intercept</th>    <td> 1.483e+05</td> <td> 1.95e+05</td> <td>    0.759</td> <td> 0.454</td> <td>-2.51e+05</td> <td> 5.47e+05</td>
</tr>
<tr>
  <th>Unemployment</th> <td>-4137.2826</td> <td> 4008.558</td> <td>   -1.032</td> <td> 0.310</td> <td>-1.23e+04</td> <td> 4049.285</td>
</tr>
<tr>
  <th>CPI_all</th>      <td> -517.9910</td> <td>  808.259</td> <td>   -0.641</td> <td> 0.526</td> <td>-2168.676</td> <td> 1132.694</td>
</tr>
<tr>
  <th>CPI_energy</th>   <td>   54.1833</td> <td>  114.076</td> <td>    0.475</td> <td> 0.638</td> <td> -178.790</td> <td>  287.157</td>
</tr>
<tr>
  <th>Queries</th>      <td>   21.1855</td> <td>   11.978</td> <td>    1.769</td> <td> 0.087</td> <td>   -3.278</td> <td>   45.649</td>
</tr>
<tr>
  <th>Month</th>        <td>  110.6853</td> <td>  191.657</td> <td>    0.578</td> <td> 0.568</td> <td> -280.731</td> <td>  502.102</td>
</tr>
</table>
<table class="simpletable">
<tr>
  <th>Omnibus:</th>       <td> 1.544</td> <th>  Durbin-Watson:     </th> <td>   1.158</td>
</tr>
<tr>
  <th>Prob(Omnibus):</th> <td> 0.462</td> <th>  Jarque-Bera (JB):  </th> <td>   1.312</td>
</tr>
<tr>
  <th>Skew:</th>          <td> 0.453</td> <th>  Prob(JB):          </th> <td>   0.519</td>
</tr>
<tr>
  <th>Kurtosis:</th>      <td> 2.764</td> <th>  Cond. No.          </th> <td>1.49e+05</td>
</tr>
</table>

In [162]:
model2.params

Intercept       148330.487698
Unemployment     -4137.282559
CPI_all           -517.991036
CPI_energy          54.183317
Queries             21.185516
Month              110.685270
dtype: float64

###Problem 4.1
Re-run the regression with the Month variable modeled as a factor variable, what is the new R-squared?

In [166]:
model3 = sm.ols(formula="ElantraSales ~ Unemployment + CPI_all + "
                        "CPI_energy + Queries + C(Month)",
data=elantra_train).fit()
model3.summary()

<table class="simpletable">
<caption>OLS Regression Results</caption>
<tr>
  <th>Dep. Variable:</th>      <td>ElantraSales</td>   <th>  R-squared:         </th> <td>   0.819</td>
</tr>
<tr>
  <th>Model:</th>                   <td>OLS</td>       <th>  Adj. R-squared:    </th> <td>   0.684</td>
</tr>
<tr>
  <th>Method:</th>             <td>Least Squares</td>  <th>  F-statistic:       </th> <td>   6.044</td>
</tr>
<tr>
  <th>Date:</th>             <td>Tue, 21 Mar 2017</td> <th>  Prob (F-statistic):</th> <td>0.000147</td>
</tr>
<tr>
  <th>Time:</th>                 <td>08:38:32</td>     <th>  Log-Likelihood:    </th> <td> -319.26</td>
</tr>
<tr>
  <th>No. Observations:</th>      <td>    36</td>      <th>  AIC:               </th> <td>   670.5</td>
</tr>
<tr>
  <th>Df Residuals:</th>          <td>    20</td>      <th>  BIC:               </th> <td>   695.9</td>
</tr>
<tr>
  <th>Df Model:</th>              <td>    15</td>      <th>                     </th>     <td> </td>   
</tr>
<tr>
  <th>Covariance Type:</th>      <td>nonrobust</td>    <th>                     </th>     <td> </td>   
</tr>
</table>
<table class="simpletable">
<tr>
         <td></td>           <th>coef</th>     <th>std err</th>      <th>t</th>      <th>P>|t|</th>  <th>[0.025</th>    <th>0.975]</th>  
</tr>
<tr>
  <th>Intercept</th>      <td> 3.125e+05</td> <td> 1.44e+05</td> <td>    2.169</td> <td> 0.042</td> <td>  1.2e+04</td> <td> 6.13e+05</td>
</tr>
<tr>
  <th>C(Month)[T.2]</th>  <td> 2254.9978</td> <td> 1943.249</td> <td>    1.160</td> <td> 0.260</td> <td>-1798.548</td> <td> 6308.543</td>
</tr>
<tr>
  <th>C(Month)[T.3]</th>  <td> 6696.5568</td> <td> 1991.635</td> <td>    3.362</td> <td> 0.003</td> <td> 2542.080</td> <td> 1.09e+04</td>
</tr>
<tr>
  <th>C(Month)[T.4]</th>  <td> 7556.6074</td> <td> 2038.022</td> <td>    3.708</td> <td> 0.001</td> <td> 3305.368</td> <td> 1.18e+04</td>
</tr>
<tr>
  <th>C(Month)[T.5]</th>  <td> 7420.2490</td> <td> 1950.139</td> <td>    3.805</td> <td> 0.001</td> <td> 3352.331</td> <td> 1.15e+04</td>
</tr>
<tr>
  <th>C(Month)[T.6]</th>  <td> 9215.8326</td> <td> 1995.230</td> <td>    4.619</td> <td> 0.000</td> <td> 5053.856</td> <td> 1.34e+04</td>
</tr>
<tr>
  <th>C(Month)[T.7]</th>  <td> 9929.4644</td> <td> 2238.800</td> <td>    4.435</td> <td> 0.000</td> <td> 5259.409</td> <td> 1.46e+04</td>
</tr>
<tr>
  <th>C(Month)[T.8]</th>  <td> 7939.4474</td> <td> 2064.629</td> <td>    3.845</td> <td> 0.001</td> <td> 3632.706</td> <td> 1.22e+04</td>
</tr>
<tr>
  <th>C(Month)[T.9]</th>  <td> 5013.2866</td> <td> 2010.745</td> <td>    2.493</td> <td> 0.022</td> <td>  818.946</td> <td> 9207.627</td>
</tr>
<tr>
  <th>C(Month)[T.10]</th> <td> 2500.1838</td> <td> 2084.057</td> <td>    1.200</td> <td> 0.244</td> <td>-1847.083</td> <td> 6847.451</td>
</tr>
<tr>
  <th>C(Month)[T.11]</th> <td> 3238.9315</td> <td> 2397.231</td> <td>    1.351</td> <td> 0.192</td> <td>-1761.605</td> <td> 8239.468</td>
</tr>
<tr>
  <th>C(Month)[T.12]</th> <td> 5293.9107</td> <td> 2228.310</td> <td>    2.376</td> <td> 0.028</td> <td>  645.738</td> <td> 9942.083</td>
</tr>
<tr>
  <th>Unemployment</th>   <td>-7739.3814</td> <td> 2968.747</td> <td>   -2.607</td> <td> 0.017</td> <td>-1.39e+04</td> <td>-1546.683</td>
</tr>
<tr>
  <th>CPI_all</th>        <td>-1343.3068</td> <td>  592.919</td> <td>   -2.266</td> <td> 0.035</td> <td>-2580.114</td> <td> -106.500</td>
</tr>
<tr>
  <th>CPI_energy</th>     <td>  288.6314</td> <td>   97.974</td> <td>    2.946</td> <td> 0.008</td> <td>   84.262</td> <td>  493.001</td>
</tr>
<tr>
  <th>Queries</th>        <td>   -4.7636</td> <td>   12.938</td> <td>   -0.368</td> <td> 0.717</td> <td>  -31.752</td> <td>   22.224</td>
</tr>
</table>
<table class="simpletable">
<tr>
  <th>Omnibus:</th>       <td> 0.047</td> <th>  Durbin-Watson:     </th> <td>   2.795</td>
</tr>
<tr>
  <th>Prob(Omnibus):</th> <td> 0.977</td> <th>  Jarque-Bera (JB):  </th> <td>   0.246</td>
</tr>
<tr>
  <th>Skew:</th>          <td>-0.032</td> <th>  Prob(JB):          </th> <td>  

### Problem 5.1
Which variables are correlated with CPI_energy, consumer price index for energy, correlation > 0.6? Treat month as numeric


In [119]:
elantra_train.corr()

<div>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Month</th>
      <th>Year</th>
      <th>ElantraSales</th>
      <th>Unemployment</th>
      <th>Queries</th>
      <th>CPI_energy</th>
      <th>CPI_all</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>Month</th>
      <td>1.000000</td>
      <td>0.000000</td>
      <td>0.109795</td>
      <td>-0.203603</td>
      <td>0.015844</td>
      <td>0.176020</td>
      <td>0.266788</td>
    </tr>
    <tr>
      <th>Year</th>
      <td>0.000000</td>
      <td>1.000000</td>
      <td>0.587274</td>
      <td>-0.958746</td>
      <td>0.726531</td>
      <td>0.831605</td>
      <td>0.948585</td>
    </tr>
    <tr>
      <th>ElantraSales</th>
      <td>0.109795</td>
      <td>0.587274</td>
      <td>1.000000</td>
      <td>-0.567146</td>
      <td>0.610065</td>
      <td>0.591649</td>
      <td>0.593622</td>
    </tr>
    <tr>
      <th>Unemployment</th>
      <td>-0.203603</td>
      <td>-0.958746</td>
      <td>-0.567146</td>
      <td>1.000000</td>
      <td>-0.641109</td>
      <td>-0.800719</td>
      <td>-0.956212</td>
    </tr>
    <tr>
      <th>Queries</th>
      <td>0.015844</td>
      <td>0.726531</td>
      <td>0.610065</td>
      <td>-0.641109</td>
      <td>1.000000</td>
      <td>0.832838</td>
      <td>0.753673</td>
    </tr>
    <tr>
      <th>CPI_energy</th>
      <td>0.176020</td>
      <td>0.831605</td>
      <td>0.591649</td>
      <td>-0.800719</td>
      <td>0.832838</td>
      <td>1.000000</td>
      <td>0.913226</td>
    </tr>
    <tr>
      <th>CPI_all</th>
      <td>0.266788</td>
      <td>0.948585</td>
      <td>0.593622</td>
      <td>-0.956212</td>
      <td>0.753673</td>
      <td>0.913226</td>
      <td>1.000000</td>
    </tr>
  </tbody>
</table>
</div>

CPI_energy is correlated with "Unemployment", "CPI_all" and "Queries" (also with "year")

### Problem 5.2
Which variables are correlated with Queries (approximation of the number of Google searches) correlation > 0.6? Treat month as numeric

Queries is correlated with "Unemployment", "CPI_all" and "CPI_energy" (also with "year")

### Problem 6.1
iteratively remove variables with high p-vales (above 0.1). Which variables and in which order are removed?

Based on "model3" above, we first remove the "Queries" variable (p-value = 0.717)

In [120]:
model4 = sm.ols(formula="ElantraSales ~ Unemployment + CPI_all +"
                        " CPI_energy + C(Month)", 
                data=elantra_train).fit()
model4.summary()

<table class="simpletable">
<caption>OLS Regression Results</caption>
<tr>
  <th>Dep. Variable:</th>      <td>ElantraSales</td>   <th>  R-squared:         </th> <td>   0.818</td>
</tr>
<tr>
  <th>Model:</th>                   <td>OLS</td>       <th>  Adj. R-squared:    </th> <td>   0.697</td>
</tr>
<tr>
  <th>Method:</th>             <td>Least Squares</td>  <th>  F-statistic:       </th> <td>   6.744</td>
</tr>
<tr>
  <th>Date:</th>             <td>Tue, 21 Mar 2017</td> <th>  Prob (F-statistic):</th> <td>5.73e-05</td>
</tr>
<tr>
  <th>Time:</th>                 <td>07:38:29</td>     <th>  Log-Likelihood:    </th> <td> -319.38</td>
</tr>
<tr>
  <th>No. Observations:</th>      <td>    36</td>      <th>  AIC:               </th> <td>   668.8</td>
</tr>
<tr>
  <th>Df Residuals:</th>          <td>    21</td>      <th>  BIC:               </th> <td>   692.5</td>
</tr>
<tr>
  <th>Df Model:</th>              <td>    14</td>      <th>                     </th>     <td> </td>   
</tr>
<tr>
  <th>Covariance Type:</th>      <td>nonrobust</td>    <th>                     </th>     <td> </td>   
</tr>
</table>
<table class="simpletable">
<tr>
         <td></td>           <th>coef</th>     <th>std err</th>      <th>t</th>      <th>P>|t|</th>  <th>[0.025</th>    <th>0.975]</th>  
</tr>
<tr>
  <th>Intercept</th>      <td> 3.257e+05</td> <td> 1.37e+05</td> <td>    2.384</td> <td> 0.027</td> <td> 4.16e+04</td> <td>  6.1e+05</td>
</tr>
<tr>
  <th>C(Month)[T.2]</th>  <td> 2410.9137</td> <td> 1857.103</td> <td>    1.298</td> <td> 0.208</td> <td>-1451.144</td> <td> 6272.972</td>
</tr>
<tr>
  <th>C(Month)[T.3]</th>  <td> 6880.0868</td> <td> 1888.145</td> <td>    3.644</td> <td> 0.002</td> <td> 2953.474</td> <td> 1.08e+04</td>
</tr>
<tr>
  <th>C(Month)[T.4]</th>  <td> 7697.3580</td> <td> 1960.214</td> <td>    3.927</td> <td> 0.001</td> <td> 3620.869</td> <td> 1.18e+04</td>
</tr>
<tr>
  <th>C(Month)[T.5]</th>  <td> 7444.6447</td> <td> 1908.477</td> <td>    3.901</td> <td> 0.001</td> <td> 3475.749</td> <td> 1.14e+04</td>
</tr>
<tr>
  <th>C(Month)[T.6]</th>  <td> 9223.1343</td> <td> 1953.636</td> <td>    4.721</td> <td> 0.000</td> <td> 5160.325</td> <td> 1.33e+04</td>
</tr>
<tr>
  <th>C(Month)[T.7]</th>  <td> 9602.7221</td> <td> 2012.661</td> <td>    4.771</td> <td> 0.000</td> <td> 5417.165</td> <td> 1.38e+04</td>
</tr>
<tr>
  <th>C(Month)[T.8]</th>  <td> 7919.4990</td> <td> 2020.993</td> <td>    3.919</td> <td> 0.001</td> <td> 3716.614</td> <td> 1.21e+04</td>
</tr>
<tr>
  <th>C(Month)[T.9]</th>  <td> 5074.2910</td> <td> 1962.230</td> <td>    2.586</td> <td> 0.017</td> <td>  993.611</td> <td> 9154.971</td>
</tr>
<tr>
  <th>C(Month)[T.10]</th> <td> 2724.2422</td> <td> 1951.778</td> <td>    1.396</td> <td> 0.177</td> <td>-1334.702</td> <td> 6783.186</td>
</tr>
<tr>
  <th>C(Month)[T.11]</th> <td> 3665.0765</td> <td> 2055.659</td> <td>    1.783</td> <td> 0.089</td> <td> -609.900</td> <td> 7940.053</td>
</tr>
<tr>
  <th>C(Month)[T.12]</th> <td> 5643.1903</td> <td> 1974.363</td> <td>    2.858</td> <td> 0.009</td> <td> 1537.278</td> <td> 9749.103</td>
</tr>
<tr>
  <th>Unemployment</th>   <td>-7971.3395</td> <td> 2840.795</td> <td>   -2.806</td> <td> 0.011</td> <td>-1.39e+04</td> <td>-2063.584</td>
</tr>
<tr>
  <th>CPI_all</th>        <td>-1377.5816</td> <td>  573.387</td> <td>   -2.403</td> <td> 0.026</td> <td>-2570.005</td> <td> -185.158</td>
</tr>
<tr>
  <th>CPI_energy</th>     <td>  268.0299</td> <td>   78.752</td> <td>    3.403</td> <td> 0.003</td> <td>  104.257</td> <td>  431.803</td>
</tr>
</table>
<table class="simpletable">
<tr>
  <th>Omnibus:</th>       <td> 0.000</td> <th>  Durbin-Watson:     </th> <td>   2.771</td>
</tr>
<tr>
  <th>Prob(Omnibus):</th> <td> 1.000</td> <th>  Jarque-Bera (JB):  </th> <td>   0.137</td>
</tr>
<tr>
  <th>Skew:</th>          <td>-0.001</td> <th>  Prob(JB):          </th> <td>   0.934</td>
</tr>
<tr>
  <th>Kurtosis:</th>      <td> 2.698</td> <th>  Cond. No.          </th> <td>1.18e+05</td>
</tr>
</table>

The p-vales for model4 are all good (except for months, but that's categorical)

### Problem 6.2
what is SSE for the prediction data with model4?

In [122]:
predict4 = model4.predict(elantra_test)

In [135]:
# calculating the SSE
SSE = sum((elantra_test['ElantraSales'].values - predict4)**2)
SSE

190757747.44431755

### Problem 6.3
What would the baseline method predict for all observations in the test set?

In [139]:
elantra_train['ElantraSales'].mean()

14462.25

### Problem 6.4
What is the test set R-Squared?

In [144]:
SST = sum((elantra_train['ElantraSales'].mean()-elantra_test['ElantraSales'])**2)
1 - SSE/SST  # Rsquared

0.72802322762912197

### Problem 6.5
What is the maximum absolute error of the predicted values?

In [146]:
(elantra_test['ElantraSales'].values - predict4).abs().max()

13

### Problem 6.6
Month and year of maximum error

In [150]:
max_index = (elantra_test['ElantraSales'].values - predict4).abs().idxmax()
elantra_test.loc[max_index][['Month', 'Year']]

Month       3.0
Year     2013.0
Name: 13, dtype: float64